pinot-core/src/test/java/org/apache/pinot/queries/TransformQueriesTest.java

```diff
@@ -0,0 +1,286 @@
+import org.apache.pinot.core.data.readers.RecordReader;
+import org.apache.pinot.core.indexsegment.IndexSegment;
+import org.apache.pinot.core.indexsegment.generator.SegmentGeneratorConfig;
+import org.apache.pinot.core.indexsegment.immutable.ImmutableSegment;
+import org.apache.pinot.core.indexsegment.immutable.ImmutableSegmentLoader;
+import org.apache.pinot.core.operator.blocks.IntermediateResultsBlock;
+import org.apache.pinot.core.operator.query.AggregationOperator;
+import org.apache.pinot.core.query.aggregation.function.customobject.AvgPair;
+import org.apache.pinot.core.segment.creator.impl.SegmentIndexCreationDriverImpl;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+public class TransformQueriesTest extends BaseQueriesTest {
+
+  private static final String D1 = "STRING_COL";
+  private static final String M1 = "INT_COL1";
+  private static final String M2 = "INT_COL2";
+  private static final String M3 = "LONG_COL1";
+  private static final String M4 = "LONG_COL2";
+  private static final String TIME = "T";
+  private static final String TABLE_NAME = "FOO";
+  private static final String SEGMENT_NAME_1 = "FOO_SEGMENT_1";
+  private static final String SEGMENT_NAME_2 = "FOO_SEGMENT_2";
+
+  private static final File INDEX_DIR = new File(FileUtils.getTempDirectory(), "TransformQueriesTest");
+
+  private Schema _schema;
+  private List<IndexSegment> _indexSegments = new ArrayList<>();
+  private List<SegmentDataManager> _segmentDataManagers;
+
+  @BeforeClass
+  public void setup() {
+    _schema = new Schema();
+
+    _schema.setSchemaName(TABLE_NAME);
+
+    final DimensionFieldSpec d1Spec = new DimensionFieldSpec(D1, FieldSpec.DataType.STRING, true);
+    _schema.addField(d1Spec);
+
+    final MetricFieldSpec m1Spec = new MetricFieldSpec(M1, FieldSpec.DataType.INT);
+    _schema.addField(m1Spec);
+
+    final MetricFieldSpec m2Spec = new MetricFieldSpec(M2, FieldSpec.DataType.INT);
+    _schema.addField(m2Spec);
+
+    final MetricFieldSpec m3Spec = new MetricFieldSpec(M3, FieldSpec.DataType.LONG);
+    _schema.addField(m3Spec);
+
+    final MetricFieldSpec m4Spec = new MetricFieldSpec(M4, FieldSpec.DataType.LONG);
+    _schema.addField(m4Spec);
+
+    final TimeFieldSpec tSpec = new TimeFieldSpec(new TimeGranularitySpec(FieldSpec.DataType.LONG, TimeUnit.MILLISECONDS, TIME));
+    _schema.addField(tSpec);
+  }
+
+  @AfterClass
+  public void destroy() {
+    FileUtils.deleteQuietly(INDEX_DIR);
+  }
+
+  private void delete() {
+    for (IndexSegment indexSegment : _indexSegments) {
+      if (indexSegment != null) {
+        indexSegment.destroy();
+      }
+    }
+    _indexSegments.clear();
+  }
+
+  @Test
+  public void testTransformWithAvgInnerSegment() throws Exception {
+    try {
+      final List<GenericRow> rows = createDataSet(10);
+
+      final RecordReader recordReader = new GenericRowRecordReader(rows, _schema);
```

Comment:
Resource leak, use try-with resource

Reply:
Changed to try with resource but in this case we were not leaking resource as record reader.close() is a NO-OP